src/components/FileMenu/DeleteDialog.js

```diff
@@ -0,0 +1,83 @@
+import React from 'react'
+
+import PropTypes from '@dhis2/prop-types'
+import i18n from '@dhis2/d2-i18n'
+import { useDataMutation } from '@dhis2/app-runtime'
+import {
+    Modal,
+    ModalTitle,
+    ModalContent,
+    ModalActions,
+    ButtonStrip,
+    Button,
+} from '@dhis2/ui'
+
+import { supportedFileTypes, endpointFromFileType } from './utils'
+
+const getMutation = type => ({
+    resource: endpointFromFileType(type),
+    id: ({ id }) => id,
+    type: 'delete',
+})
+/*
+const mapMutation = {
+    resource: 'maps',
+    id: ({ id }) => id,
+    type: 'delete',
+}
+
+const visualizationMutation = {
+    resource: 'visualizations',
+    id: ({ id }) => id,
+    type: 'delete',
+}
+*/
+export const DeleteDialog = ({ type, id, onClose, onDelete, onError }) => {
+    const [mutate, { error }] = useDataMutation(
+        //        type === 'map' ? mapMutation : visualizationMutation,
+        getMutation(type)
```

Comment:
@amcgee using it this way I get a warning about the mutation not being static.
I guess I didn't understand what we said on Slack about using a function.

Reply:
Try this - it's a tad hacky but works fine:

```
const mutation = useMemo(getMutation(type), [type])
const [mutate, { error }] = useDataMutation(mutation)
```

Otherwise, the "fully static" way to do it would be to have a separate const mutation object for each resource type.

Reply:
(you could even exclude the `[type]` dependency and leave it `[]` to explicitly show that the mutation won't change if the type prop changes - this is what the `useStaticInput` hook does internally when calling the mutation

The "fully static" mechanism would be best since it supports static analysis, but `useMemo` is OK for now

Reply:
Fixed as suggested.